app/models/git_repository.rb

```diff
@@ -70,9 +71,14 @@ def self_signed_cert_cb(_valid, _host)
     true
   end
 
+  # NOTE:  Please make sure to use this whenever you implement a method that is
+  # modifying the `.git` data (clone, fetch, checking, etc.) so that we can be
+  # sure this happens safely between processes.
   def with_worktree
-    handling_worktree_errors do
-      yield worktree
+    git_transaction do
```

Comment:
So this one should be dropped, IMO.

Reply:
I still don't think this should lock here as this effectively serializes every operation, and we only want to serialize the clones